### PR TITLE
issue#1081 -- documented regexp_replace function

### PR DESCRIPTION
**Pull request by AnkitaMalik22**
#1081

### CODE REVIEW

File: docs/reference/operators/regex.md [hunk 1]
```diff
@@ -20,3 +20,9 @@ matching via regular expressions.
 `string !~ regex` - checks if `string` value does not match regex
 
 `symbol !~ regex` - checks if `symbol` value does not match regex
+
+## regexp_replace (replace)
+
+`string regexp_replace regex` -  replaces a sequence of characters in a string with another set of characters using regular expression pattern matching
+
+`symbol regexp_replace regex` - replaces `symbol` value to regex
```
Comment (amyshwang) [inline docs/reference/operators/regex.md:28]: This does not seem to be a complete description?

File: docs/reference/operators/regex.md [hunk 2]
```diff
@@ -20,3 +20,9 @@ matching via regular expressions.
 `string !~ regex` - checks if `string` value does not match regex
 
 `symbol !~ regex` - checks if `symbol` value does not match regex
+
+## regexp_replace (replace)
+
+`string regexp_replace regex` -  replaces a sequence of characters in a string with another set of characters using regular expression pattern matching
```
Comment (amyshwang) [inline docs/reference/operators/regex.md:26]: Full stop missing - it is also missing from the description above; would you mind adding them too please?

**Review by amyshwang — CHANGES_REQUESTED** [review]
Thank you for your contribution. However, as described in the [guide](https://github.com/questdb/questdb.io/issues/1085), please:
* Write in the issue you would like to take and say that you would like to take it on.

In addition, although other functions on this page do not have examples, we'd like to include some examples using this function, as described by the function template.

I will be able to review the PR again once it has been updated. Thank you again!

File: docs/reference/operators/regex.md [hunk 3]
```diff
@@ -20,3 +20,30 @@ matching via regular expressions.
...
 
 `symbol !~ regex` - checks if `symbol` value does not match regex
+
+---
+title: Regex functions
+sidebar_label: Regex
+description: Regex functions reference documentation.
+---
+
+Regex functions will allow us to replace a sequence of characters in a string with another set of characters using regular expression pattern matching.
```
Comment (amyshwang) [inline docs/reference/operators/regex.md:30]: This is only needed on top of each page. Please delete.

File: docs/reference/operators/regex.md [hunk 4]
```diff
@@ -20,3 +20,30 @@ matching via regular expressions.
...
+
+## regexp_replace (replace)
```
Comment (amyshwang) [inline docs/reference/operators/regex.md:32]: Consistency:

```suggestion
## regexp_replace
```

Could you please ensure that the section follows the pattern similar to:
image

File: docs/reference/operators/regex.md [hunk 5]
```diff
@@ -20,3 +20,30 @@ matching via regular expressions.
...
+
+**Return value:**
+
+The source string is returned unchanged if there is no match to the pattern. If there is a match, the source string is returned with the replacement string substituted for the matching substring.
+
+**Examples:**
+
+```questdb-sql title="Example description -  regexp_replace"
+SELECT REGEXP_REPLACE ('MYSQL is a great database', '^(\S*)', 'QuestDB')
+FROM dual;
```
Comment (amyshwang) [inline docs/reference/operators/regex.md:48]:
```suggestion
SELECT REGEXP_REPLACE ('MYSQL is a great database', '^(\S*)', 'QuestDB');
```

**Review by amyshwang — CHANGES_REQUESTED** [review]
Thank you! Editorial review comments added.

File: docs/reference/operators/regex.md [hunk 6]
```diff
@@ -20,3 +20,30 @@ matching via regular expressions.
...
+title: Regex functions
+sidebar_label: Regex
+description: Regex functions reference documentation.
+---
+
+Regex functions will allow us to replace a sequence of characters in a string with another set of characters using regular expression pattern matching.
+
+## regexp_replace (replace)
+
+`string regexp_replace regex` - The regexp_replace function provides substitution of new text for substrings that match POSIX regular expression patterns.
```
Comment (puzpuzpuz) [inline docs/reference/operators/regex.md:34]: We use Java regex syntax, not POSIX, and it's worth mentioning that.

File: docs/reference/operators/regex.md [hunk 7]
```diff
@@ -20,3 +20,30 @@ matching via regular expressions.
...
+
+Regex functions will allow us to replace a sequence of characters in a string with another set of characters using regular expression pattern matching.
+
+## regexp_replace (replace)
+
+`string regexp_replace regex` - The regexp_replace function provides substitution of new text for substrings that match POSIX regular expression patterns. 
+
+**Arguments:**
+
+- regexp_replace ( string text, pattern text, replacement text [, flags text ] )
```
Comment (puzpuzpuz) [inline docs/reference/operators/regex.md:38]: We don't support flag. Instead, we default to global case sensitive replacement.

File: docs/reference/operators/regex.md [hunk 8]
```diff
@@ -20,3 +20,25 @@ matching via regular expressions.
 `string !~ regex` - checks if `string` value does not match regex
 
 `symbol !~ regex` - checks if `symbol` value does not match regex
+
+
+
+## regexp_replace
+
+`regexp_replace ( value1, text , value2 )` - The regexp_replace function provides substitution of new text for substrings that match regular expression patterns.
```
Comment (amyshwang) [inline docs/reference/operators/regex.md:28]:
```suggestion
`regexp_replace ( value1, text , value2 )` - provides substitution of new text for substrings that match regular expression patterns. 
```

File: docs/reference/operators/regex.md [hunk 9]
```diff
@@ -20,3 +20,25 @@ matching via regular expressions.
...
+
+## regexp_replace
+
+`regexp_replace ( value1, text , value2 )` - The regexp_replace function provides substitution of new text for substrings that match regular expression patterns. 
+
+**Arguments:**
+
+- `value1` is string value.
+- `text` is  any regular expression pattern.
+- `value2` is string value.
```
Comment (amyshwang) [inline docs/reference/operators/regex.md:34]:
```suggestion
- `value1` is any `string` value.
- `text` is  any regular expression pattern.
- `value2` is any `string` value.
```

**Review by amyshwang — CHANGES_REQUESTED** [review]
Minor editorial review.

**Review by amyshwang — CHANGES_REQUESTED** [review]
A couple more comments - I think it is almost ready!

**Review by amyshwang — APPROVED** [review]
Thank you for your contribution! 🚀

**Pull request merged**